src/Features/LanguageServer/ProtocolUnitTests/Workspaces/LspWorkspaceManagerTests.cs

```diff
@@ -65,12 +67,25 @@ public async Task TestLspUsesWorkspaceInstanceOnChangesAsync()
         var firstDocumentInitialVersion = await firstDocument.GetSyntaxVersionAsync(CancellationToken.None);
         var secondDocumentInitialVersion = await secondDocument.GetSyntaxVersionAsync(CancellationToken.None);
 
-        // Verify the LSP documents are the same instance as the workspaces documents.
-        Assert.Same(testLspServer.TestWorkspace.CurrentSolution.GetDocument(firstDocument.Id), firstDocument);
+        if (mutatingLspWorkspace)
+        {
+            // In the mutating case, adding/opening the second document will cause the workspace solution to actually
+            // change.  So it will not point to a different document instance for firstDocument.   The underlying state
```

Comment:
did you mean this?
```suggestion
            // change.  So it will point to a different document instance for firstDocument.   The underlying state
```

Reply:
yes.  quite right :)